### ISSUE TITLE
fromIndex and toIndex coming as -1 when trying to drag and drop

### ISSUE THREAD

**Issue by rajeev-tripathi** [cdwv/ui-grid-draggable-rows#41]
Hi,
I have basically two grids on the same page. I have some sequencing logic which works perfectly for one grid. 
I wanted the same logic to be applied on my 2nd grid. Thus, called a common function onDragEnd and passed my table name to segregate my logic.

E.g.
//First Grid
var grid1Options = {
...
..
 onRegisterApi: function () {
    gridApi.dragRows.on.dragEnd($scope, function (draggableObject) {
          onDragEnd('grid1', draggableObject);
        });
 }
}

// Second Grid
var grid2Options = {
...
..
 onRegisterApi: function () {
    gridApi.dragRows.on.dragEnd($scope, function (draggableObject) {
          onDragEnd('grid2', draggableObject);
        });
 }
}

// my common function
function onDragEnd(sourceTable, draggableObject) {
// my logic here works perfectly for grid1 but for grid2 draggableObject gives me fromIndex and toIndex as -1
}

**Comment by bazizten** [comment]
I have the same issue. Getting -1 as the toIndex. If I try a couple of times more it will work sooner or later.

**Comment by skrajewski** [comment]
Could you give a working example, for example in codepen or pastebin?

**Comment by rajeev-tripathi** [comment]
Apologies for posting my response late. I figured out my issue. Since there were two Grids and I had the same method for both the grids _onDragEnd_ 
The indexOf method was returning the same index.
function indexOf(entity) {
    return _gridObject.data.indexOf(entity)_
}
I modified above method as 

function indexOf(entity) {
   if(myActiveGrid === 'grid1') {
     return grid1.data.indexOf(entity);
   }
  return grid2.data.indexOf(entity);
}
